### DISCUSSION TITLE
Multi-tenant Postgres DB Performance

### DISCUSSION THREAD

**Discussion by besh0y** [hasura/graphql-engine#7206] [General] [answered]
**Hello there!**
I'm planning to use multi-tenant architecture in my app and I'm wondering if having 10k+ schemas in my DB would affect Hasura's performance by any means including migrations and metadata application.

**Comment by Adron** [comment, marked as answer]
Hey @besh0y! This would definitely effect performance, but primarily from the migrations and metadata perspective. The queries, and related interactions with the schemas once they're on your API/server/databases would work based on how they're structured.

However, my suggestion would be to limit the amount of schemas when there are that many. Most specifically limit them to the use cases expected from the GraphQL API calls.
